sys/linux/bpf.txt

```diff
@@ -268,6 +273,7 @@ link_create_arg_extra [
 	netfilter	link_create_netfilter
 	tcx		link_create_tcx
 	uprobe_multi	link_create_uprobe_multi
+	netkit		link_create_netkit
```

Comment:
Do I understand it correctly that, with how descriptions are currently written, syzkaller needs to get super lucky for `BPF_LINK_TYPE_NETKIT` and the `netkit` field of this union to coincide in a single proram?

Reply:
It's actually demultiplexed not based on `BPF_LINK_TYPE_XXX` but based on program types, derived from the `link_create.prog_fd` (defined just below): https://elixir.bootlin.com/linux/v6.6/source/kernel/bpf/syscall.c#L4905. But yes, same story: it will be difficult for syzkaller to reach this.

We'll have to break this out by program types at some point, but it's a bit painful to do (very verbose), so I keep postponing :disappointed: We have the same issues for several other bpf(2) commands who also depend on the program type (or on map types). https://github.com/google/syzkaller/issues/2226 would likely help a lot here.

Reply:
I see :(

I'll try to boost priority for #2226. Hopefully it shouldn't actually be too hard to implement (at least if focus on simple conditions like `X&FLAG > 0` and `X==CONST` first).